src/main/java/com/example/petbuddybackend/service/care/CareService.java

```diff
@@ -8,177 +8,108 @@
-
-    }
+        userService.assertNotBlockedByAny(clientEmail, caretakerEmail);
 
-    public CareDTO updateCare(Long careId, UpdateCareDTO updateCare, String userEmail, ZoneId timeZone) {
+        Set<AnimalAttribute> animalAttributes = animalService.getAnimalAttributes(createCare.animalAttributeIds());
+        assertAnimalAttributesMatchAnimalType(animalAttributes, createCare.animalType());
 
-        Care care = getCare(careId);
-        assertLoggedInUserIsCaretaker(care.getCaretaker().getEmail(), userEmail, "Only caretaker can edit care");
-        assertCareIsNotTerminated(care);
-        assertCaretakerStatusIsPending(care, "Cannot update accepted care");
-        assertEndCareDateIsAfterStartCareDate(updateCare.careStart(), updateCare.careEnd());
-        careMapper.updateCareFromDTO(updateCare, care);
-        care.setClientStatus(CareStatus.PENDING);
-        Care savedCare = careRepository.save(care);
-        notificationService.addNotificationForClientAndSend(
-                savedCare.getId(),
-                ObjectType.CARE,
-                savedCare.getClient(),
-                "message.care.update_reservation",
-                Set.of(care.getCaretaker().getEmail())
-        );
-        return careMapper.mapToCareDTO(savedCare, timeZone);
+        Care care = careRepository.save(
+                createCareFromReservation(clientEmail, caretakerEmail, createCare, animalAttributes));
 
+        sendCaretakerCareNotification(care, RESERVATION_MESSAGE);
+        return careMapper.mapToCareDTO(care, timeZone);
     }
 
-    public CareDTO acceptCareByCaretaker(Long careId, String userEmail, ZoneId timeZone) {
-
-        Care care = getCare(careId);
-        assertLoggedInUserIsCaretaker(care.getCaretaker().getEmail(), userEmail,
-                "Only caretaker can change caretaker status");
-        assertCareIsNotTerminated(care);
-        assertCaretakerStatusIsPending(care, "Care already accepted");
-        assertClientStatusIsAccepted(care);
-        care.setCaretakerStatus(CareStatus.AWAITING_PAYMENT);
-        care.setClientStatus(CareStatus.AWAITING_PAYMENT);
-        Care savedCare = careRepository.save(care);
-        notificationService.addNotificationForClientAndSend(
-                savedCare.getId(),
-                ObjectType.CARE,
-                savedCare.getClient(),
-                "message.care.accepted_reservation",
-                Set.of(care.getCaretaker().getEmail())
-        );
-        return careMapper.mapToCareDTO(savedCare, timeZone);
-    }
+    public CareDTO updateCare(Long careId, UpdateCareDTO updateCare, String caretakerEmail, ZoneId timeZone) {
+        Care care = getCareOfCaretaker(careId, caretakerEmail);
+        assertEndCareDateIsAfterStartCareDate(updateCare.careStart(), updateCare.careEnd());
 
-    public CareDTO acceptCareByClient(Long careId, String userEmail, ZoneId timeZone) {
+        careStateMachine.transitionToEditCare(care);
```

Comment:
nie wiem czy bym nie zostal w konwencji i nie usunac transistionToEditCare
i dal po prostu 2 razy wywolanie transistion
Jedno na accepta dla opiekuna
a drugie z accepta na pending dla klienta

Reply:
chociaz po akceptacji uruchaia sie metoda, ktora patrzy czy nie dac AWAITING_PAYMENT, to niech tak bedzie ze akceptacja to akceptacja, a przy edycji niech sie wywola da metoda transistionToEditCare, tylko tak jak wczesniej pisalem, bo tej edycji opiekun powinien miec status ACCEPTED

Reply:
> nie wiem czy bym nie zostal w konwencji i nie usunac transistionToEditCare
i dal po prostu 2 razy wywolanie transistion
Jedno na accepta dla opiekuna
a drugie z accepta na pending dla klienta

Z jednej strony chciałem uniknąć sytuacji gdzie ktoś sobie ręcznie ustawi z ACCEPT -> PENDING (bo z tego co rozumiem to nie można cofnąć akceptacji). I ta metodka to generalnie nie jest zła imo. Ta maszyn stanu w tym `transition` rozpatruje indywidualne zmiany stanu dla roli a ta funkcja dla obu ról. Plus też w tej metodzie `transitionToEditCare` sprawdzam jakie powinny być statusy przed zmianą więc raczej nie powinno się nic zepsuć

Reply:
> przy edycji niech sie wywola da metoda transistionToEditCare, tylko tak jak wczesniej pisalem, po tej edycji opiekun powinien miec status ACCEPTED

Zmieni się :+1: